### PR TITLE
Fix tabview

### LINKED ISSUE #90: 🐛 Tabs - On hover close button flickering
**Describe the bug**
Using   fluent_ui: ^3.4.0, when mouse hovers on Tabs, the "X" close button flicks from grey to white. Maybe is some side effect of enabling/disabling the close button ?

**To Reproduce**
1. Go to https://bdlukaa.github.io/fluent_ui/ 
2. Click on 'Others'
3. Scroll down to the Tabs demo
4. Pass the mouse pointer over the tabs. The close button flicks and retains a darker grey background in comparison with the rest of the Tab.
5. Move the mouse again. The close button flicks again.

**Expected behavior**
I'm not expecting anything special

**Screenshots**
N/A

**Additional context**
N/A

### LINKED ISSUE #91: Tabs - Close on middle click
**Is your feature request related to a problem? Please describe.**
Using mouse, when a Tab is pressed using the middle button, the tab is not closed as expected. Instead, the tab moves into a draggable state. This is the default behaviour in Chrome, for example.

**Describe the solution you'd like**
Middle click -> Close the tab 

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### LINKED ISSUE #92: 🐛 TabView - showScrollButtons does nothing
**Describe the bug**
In tabs, the showScrollButtons flag does noting. 

**To Reproduce**

1. Setup a "`TabView`" and set `showScrollButtons = true`
2. Compile and run
3. Create a lot of tabs
4. ScrollButtons aren't showed at any time, so offscreen tabs aren't reachable anymore.

**Expected behavior**
Have some scroll controls when the opened tabs doesn't fits the available width. 

**Screenshots**
N/A

**Additional context**
As a workarround, I've implemented a hack using the "new" button. I've transformed the "new" button in a one that shows a scrollable bottom sheet with all tabs. Every tab is focusable on tap and also Closeable using a dedicated button

Captura de pantalla 2021-11-11 a las 17 31 02

### PR DESCRIPTION
**Pull request by WinXaito**
Fixes #92 
Fixes #90 
Fixes #91 

## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files  
- [x] I have updated `CHANGELOG.md` with my changes  
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could
- [x] I have added/updated relevant documentation
- [x] I have run `flutter pub publish --dry-run` and addressed any warnings

### CODE REVIEW

**Comment by WinXaito** [comment]
There is one bug that I can't correct.

When the tabview is in a scrollable widget, the horitontal scroll on tabs doesn't prevent parent scroll. 

I searched a long moment for that and found nothing that work. So I added a parameter `wheelScroll` for enable or disable the wheel scroll on tabs (by default disabled).

File: lib/src/controls/navigation/tab_view.dart [hunk 1]
```diff
@@ -35,12 +42,20 @@ class TabView extends StatelessWidget {
...
         assert(minTabWidth > 0 && maxTabWidth > 0),
         assert(minTabWidth < maxTabWidth),
-        super(key: key);
+        super(key: key) {
+    this.scrollPosController = scrollPosController ??
+        ScrollPosController(
+          itemCount: tabs.length,
+          animationDuration: const Duration(milliseconds: 100),
+        );
+  }
```
Comment (bdlukaa) [inline lib/src/controls/navigation/tab_view.dart:58]: It'd be better to handle this in a `StatefulWidget`. Also, `scrollPosController` should be disposed

File: lib/src/controls/navigation/tab_view.dart [hunk 2]
```diff
@@ -90,6 +105,13 @@ class TabView extends StatelessWidget {
   /// should be displayed, if necessary. Defaults to true
   final bool showScrollButtons;
 
+  /// The controller used for move tabview to right and left when the
+  /// larger of all items is bigger than screen width.
+  late final ScrollPosController scrollPosController;
```
Comment (bdlukaa) [inline lib/src/controls/navigation/tab_view.dart:110]: `scrollController` is a better name, no?

File: lib/src/controls/navigation/tab_view.dart [hunk 3]
```diff
@@ -1,8 +1,15 @@
 import 'package:fluent_ui/fluent_ui.dart';
 
 import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/services.dart';
+import 'package:scroll_pos/scroll_pos.dart';
```
Comment (bdlukaa) [inline lib/src/controls/navigation/tab_view.dart:7]: `'package:scroll_pos/scroll_pos.dart'` should be exported

File: lib/src/controls/navigation/tab_view.dart [hunk 4]
```diff
@@ -90,6 +105,13 @@ class TabView extends StatelessWidget {
   /// should be displayed, if necessary. Defaults to true
   final bool showScrollButtons;
 
+  /// The controller used for move tabview to right and left when the
+  /// larger of all items is bigger than screen width.
```
Comment (bdlukaa) [inline lib/src/controls/navigation/tab_view.dart:109]:
```suggestion
  /// The [ScrollPosController] used to move tabview to right and left when the
  /// tabs don't fit the available horizontal space.
  ///
  /// If null, a [ScrollPosController] is created internally.
```

File: lib/src/controls/navigation/tab_view.dart [hunk 5]
```diff
@@ -90,6 +105,13 @@ class TabView extends StatelessWidget {
...
+  late final ScrollPosController scrollPosController;
+
+  /// Indicate if the wheel scroll change the tabs positions.
```
Comment (bdlukaa) [inline lib/src/controls/navigation/tab_view.dart:112]:
```suggestion
  /// Indicate if the wheel scroll changes the tabs positions.
  ///
  /// Defaults to `false`
```

File: lib/src/controls/navigation/tab_view.dart [hunk 6]
```diff
@@ -137,21 +159,24 @@ class TabView extends StatelessWidget {
...
-        child: _Tab(
-          tab,
-          key: ValueKey<int>(index),
-          reorderIndex: isReorderEnabled ? index : null,
-          selected: index == currentIndex,
-          onPressed: onChanged == null ? null : () => onChanged!(index),
-          animationDuration: FluentTheme.of(context).fastAnimationDuration,
-          animationCurve: FluentTheme.of(context).animationCurve,
+    final Widget child = GestureDetector(
+      onTertiaryTapUp: (_) => tabs[index].onClosed?.call(),
```
Comment (bdlukaa) [inline lib/src/controls/navigation/tab_view.dart:163]:
```suggestion
      onTertiaryTapUp: (_) => tab.onClosed?.call(),
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,7 +2,10 @@ Date format: DD/MM/YYYY
 
 ## [3.5.3] - [DD/MM/YYYY]
 
+- Fix scroll buttons when we have too many tabs in `TabView` ([#92](https://github.com/bdlukaa/fluent_ui/issues/92))
```
Comment (bdlukaa) [inline CHANGELOG.md:5]:
```suggestion
- Fix scroll buttons when there are too many tabs in `TabView` ([#92](https://github.com/bdlukaa/fluent_ui/issues/92))
```

**Review by bdlukaa — CHANGES_REQUESTED** [review]

**Review by bdlukaa — APPROVED** [review]

**Pull request merged**